src/parsing/refactored_2_1/backups/final_updated_pipeline.py

```
@@ -36,5 +34,5 @@
             progress.complete_document(doc_id, len(nodes))
 
         except Exception as e:
-            logger.error(f"Failed to process {src}: {e}")
+            logger.exception(f"Failed to process {src}: {e}")
```

Ruff (0.11.9)

Undefined name `logger`

(F821)